app/controllers/course/leaderboards_controller.rb

```diff
@@ -21,6 +22,26 @@ def load_settings
     @leaderboard_settings = component.settings
   end
 
+  # Load approved students from current course with course statistics.
+  def load_course_users
+    load_course_users_by_points
+    load_course_users_by_achievement
+  end
+
+  # Load the course users, ordered by experience points
+  def load_course_users_by_points
+    @course_users_by_points ||=
+      @course.course_users.students.with_approved_state.
+      ordered_by_experience_points.includes(:user).take(display_user_count)
```

Comment:
actually the alternative if you can pass `@course.course_users.students.with_approved_state.includes(:user)` as the common variable, then the view can just do `.ordered_by_X.take(display_user_count)` instead of needing to push so many objects over?